synapse/config/cas.py

```diff
@@ -30,13 +30,22 @@ def read_config(self, config, **kwargs):
 
         if self.cas_enabled:
             self.cas_server_url = cas_config["server_url"]
-            public_base_url = cas_config.get("service_url") or self.public_baseurl
-            if public_base_url[-1] != "/":
-                public_base_url += "/"
+
+            # The public baseurl is required because it is used by the redirect
+            # template.
+            public_baseurl = self.public_baseurl
+            if not public_baseurl:
+                raise ConfigError("cas_config requires a public_baseurl to be set")
+
+            # If the service_url is given, use it for backwards compatibility.
+            # Otherwise, use the public_baseurl.
```

Comment:
If we're requiring `public_baseurl` anyway, maybe we should just rip out support for this fallback? They should be the same anyway?

Reply:
yeah, probably.